File: WebCatalogBackend/src/main/java/com/cafebabe/model/Cpu.java
package com.cafebabe.model;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.Entity;
import javax.persistence.ManyToMany;
import javax.persistence.ManyToOne;
import java.util.List;

@Getter
@Setter
@Entity
public class Cpu extends BaseDataObject {

    protected Integer clockSpeedImMHz;
    @ManyToOne
    protected Gpu integratedGpu;

    @ManyToMany
    protected List<CpuCoresBlock> coresBlocks;

    protected Integer digitalCapacityInBits;
    protected Integer techprocessInNm;

}

File: WebCatalogFrontend/src/app/model/cpu.ts
import {Gpu} from './gpu';
import {BaseDataObject} from './base-data-object';

export class Cpu extends BaseDataObject {

  clockSpeedImMHz: number;
  integratedGpu: Gpu;
  coresBlocks: Array<CpuCoresBlock> = [];
  digitalCapacityInBits: number;
  techprocessInNm: number;
}

export class CpuCoresBlock extends BaseDataObject{
  coresAmount: number;
  clockSpeedInMHz: number;
}

export enum CpuType {

  MOBILE = 'MOBILE', DESKTOP = 'DESKTOP', NOTEBOOK = 'NOTEBOOK'
}
